jetty-websocket/websocket-core/src/test/java/org/eclipse/jetty/websocket/testimpl/WebSocketHandler.java

```diff
@@ -0,0 +1,457 @@
+    {
+        boolean upgradeRequest(Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException;
+    }
+
+    interface WebSocketConnectionFactory extends ConnectionFactory
+    {
+        WebSocketCoreConnection newConnection(Connector connector, EndPoint endPoint, List<String> extensions);
+
+        @Override
+        default Connection newConnection(Connector connector, EndPoint endPoint)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        default String getProtocol()
+        {
+            return "ws";
+        }
+
+        @Override
+        default List<String> getProtocols()
+        {
+            return Collections.singletonList(getProtocol());
+        }
+    }
+
+    interface WebSocketSessionFactory
+    {
+        // TODO why is core session abstract?
+        WebSocketCoreSession<ContainerLifeCycle,WebSocketCoreConnection,WebSocketLocalEndpoint,WebSocketRemoteEndpoint>
+        newSession(WebSocketCoreConnection connection, String subprotocol);
+    }
+
+
+    public WebSocketHandler()
+    {
+    }
+
+    @Override
+    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
+    {
+        Handshaker handshaker = getHandshaker(baseRequest);
+        if (LOG.isDebugEnabled())
+            LOG.debug("handle {} handshaker={}",baseRequest,handshaker);
+        if (handshaker!=null && handshaker.upgradeRequest(baseRequest, request, response))
+            return;
+        super.handle(target,baseRequest,request,response);
+    }
+
+
+    protected Handshaker getHandshaker(Request baseRequest)
+    {
+        // TODO this can eventually be made pluggable
+        HttpField version = baseRequest.getHttpFields().getField(HttpHeader.SEC_WEBSOCKET_VERSION);
+        if (version !=null && version.getIntValue()==RFC6455Handshaker.VERSION)
+            return RFC6455_Handshaker;
+        return null;
+    }
+
+
+    private final static class RFC6455Handshaker implements Handshaker
+    {
+
+        final static int VERSION = 13;
+
+        public boolean upgradeRequest(Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException
+        {
+            if (!HttpMethod.GET.is(request.getMethod()))
+            {
+                if (LOG.isDebugEnabled())
+                    LOG.debug("not upgraded method!=GET {}",baseRequest);
+                return false;
+            }
+
+            if (!HttpVersion.HTTP_1_1.equals(baseRequest.getHttpVersion()))
+            {
+                if (LOG.isDebugEnabled())
+                    LOG.debug("not upgraded version!=1.1 {}",baseRequest);
+                return false;
+            }
+
+            boolean upgrade = false;
+            QuotedCSV connectionCSVs = null;
+            String key = null;
+            QuotedCSV extensionCSVs = null;
```

Comment:
This class is does not work for the extension parsing.  The autobahn tests will fail when using QuotedCSV.